lib/std/tz.zig

```diff
@@ -0,0 +1,233 @@
+            if (idx > counts.charcnt - 1) return error.Malformed; // rfc8536: (desig)idx [...] Each index MUST be in the range [0, "charcnt" - 1]
+            timetypes[i] = .{
+                .offset = offset,
+                .flags = dst,
+                .name_data = undefined,
+            };
+
+            // Temporarily cache idx in name_data to be processed after we've read the designator names below
+            timetypes[i].name_data[0] = idx;
+        }
+
+        var designators_data: [256 + 6]u8 = undefined;
+        try reader.readNoEof(designators_data[0..counts.charcnt]);
+        const designators = designators_data[0..counts.charcnt];
+        if (designators[designators.len - 1] != 0) return error.Malformed; // rfc8536: charcnt [...] includes the trailing NUL (0x00) octet
+
+        // Iterate through the timetypes again, setting the designator names
+        for (timetypes) |*tt| {
+            const name = std.mem.sliceTo(designators[tt.name_data[0]..], 0);
+            // We are mandating the "SHOULD" 6-character limit so we can pack the struct better, and to conform to POSIX.
+            if (name.len > 6) return error.Malformed; // rfc8536: Time zone designations SHOULD consist of at least three (3) and no more than six (6) ASCII characters.
+            std.mem.copy(u8, tt.name_data[0..], name);
+            tt.name_data[name.len] = 0;
+        }
+
+        // Parse leap seconds
+        i = 0;
+        while (i < counts.leapcnt) : (i += 1) {
+            const occur = try reader.readIntBig(i64);
+            if (occur < 0) return error.Malformed; // rfc8536: occur [...] MUST be nonnegative
+            if (i > 0 and leapseconds[i - 1].occurrence + 2419199 > occur) return error.Malformed; // rfc8536: occur [...] each later value MUST be at least 2419199 greater than the previous value
+            if (occur > std.math.maxInt(i48)) return error.Malformed; // Unreasonably far into the future
+
+            const corr = try reader.readIntBig(i32);
+            if (i == 0 and corr != -1 and corr != 1) return error.Malformed; // rfc8536: The correction value in the first leap-second record, if present, MUST be either one (1) or minus one (-1)
+            if (i > 0 and leapseconds[i - 1].correction != corr + 1 and leapseconds[i - 1].correction != corr - 1) return error.Malformed; // rfc8536: The correction values in adjacent leap-second records MUST differ by exactly one (1)
+            if (corr > std.math.maxInt(i16)) return error.Malformed; // Unreasonably large correction
+
+            leapseconds[i] = .{
+                .occurrence = @intCast(i48, occur),
+                .correction = @intCast(i16, corr),
+            };
+        }
+
+        // Parse standard/wall indicators
+        i = 0;
+        while (i < counts.isstdcnt) : (i += 1) {
+            const stdtime = try reader.readByte();
+            if (stdtime == 1) {
+                timetypes[i].flags |= 0x02;
+            }
+        }
```

Comment:
```suggestion
        for (timetypes) |*tt| {
            const stdtime = try reader.readByte();
            if (stdtime == 1) {
                tt.* |= 0x02;
            }
        }
```

Reply:
`counts.isstdcnt` is allowed to be zero, which would mess this up.